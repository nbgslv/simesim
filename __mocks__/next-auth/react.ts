const mockSession = {
  expires: new Date(Date.now() + 2 * 86400).toISOString(),
  user: {
    email: '0512345678',
    emailEmail: 'test@test.com',
    id: 'clcclcclcclcclcclc1234',
    image: null,
    name: 'foo bar',
    role: 'ADMIN',
  },
};

export const useSession = jest.fn(
  () => ({ data: mockSession, status: 'authenticated' }) // return type is [] in v3 but changed to {} in v4
);
export const signOut = jest.fn(() =>
  Promise.resolve({ url: process.env.NEXTAUTH_URL as string })
);
