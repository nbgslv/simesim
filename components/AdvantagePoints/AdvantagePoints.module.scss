@import "styles/variables";

.main {
  display: flex;
  margin-top: 2rem;
  height: 20%;

  :global svg {
    width: 3rem;
    height: 3rem;
  }

  :global * {
    font-size: 0.7rem;
  }
}

.icon {
  margin-bottom: 0.5rem;
}

.title {
  font-size: 0.875rem !important;
  font-weight: 900 !important;
  margin-bottom: 0.5rem;
}

.column {
  width: 25%;

}

.empty {
  content: '';
  width: 100%;
  height: 130px;

  @media (max-width: map-get($grid-breakpoints, 'md')) {
    height: 150px;
  }
}