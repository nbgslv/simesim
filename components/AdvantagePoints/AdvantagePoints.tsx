import React, { useEffect } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { solid } from '@fortawesome/fontawesome-svg-core/import.macro';
import { motion, useAnimationControls } from 'framer-motion';
import { Col, Container, Row } from 'react-bootstrap';
import styles from './AdvantagePoints.module.scss';

const MotionRow = motion(Row);

const AdvantagePoints = () => {
  const [message, setMessage] = React.useState<string>('');
  const controls = useAnimationControls();

  useEffect(() => {
    if (message) {
      controls.start({
        opacity: 1,
        transition: {
          duration: 0.5,
        },
      });
    } else {
      controls.start({
        opacity: 0,
        transition: {
          duration: 0.5,
        },
      });
    }
  }, [message]);

  return (
    <Container className={styles.main}>
      <Row layout="position" className="mb-2">
        <Col
          onMouseEnter={() =>
            setMessage('רכשתם חבילה והטלפון לא תומך? קבלו החזר כספי בלי שאלות')
          }
          onMouseLeave={() => setMessage('')}
          className={styles.column}
        >
          <div className={styles.icon}>
            <FontAwesomeIcon icon={solid('money-bill-transfer')} />
          </div>
          <div className={styles.title}>החזר כספי מובטח</div>
        </Col>
        <Col
          onMouseEnter={() =>
            setMessage(
              'כל הפרטים באתר מוצפנים ומועברים בצורה מאובטחת בטכנולוגיית SSL. כל עסקאות האשראי מאובטחות בתקן PCI המתקדם'
            )
          }
          onMouseLeave={() => setMessage('')}
          className={styles.column}
        >
          <div className={styles.icon}>
            <FontAwesomeIcon icon={solid('shield')} />
          </div>
          <div className={styles.title}>אבטחת המידע שלכם</div>
        </Col>
        <Col
          onMouseEnter={() =>
            setMessage(
              'מגוון חבילות חו"ל רחב במחירים הנמוכים ביותר; יותר מ-100 מדינות נתמכות; מעבר בין מדינות בלי להחליף eSim!'
            )
          }
          onMouseLeave={() => setMessage('')}
          className={styles.column}
        >
          <div className={styles.icon}>
            <FontAwesomeIcon icon={solid('shekel-sign')} />
          </div>
          <div className={styles.title}>מגוון חבילות חו&quot;ל רחב</div>
        </Col>
        <Col
          onMouseEnter={() =>
            setMessage("גישה לחבילות ותמיכה בוואצאפ, בצ'אט ובטלפון מסביב לשעון")
          }
          onMouseLeave={() => setMessage('')}
          className={styles.column}
        >
          <div className={styles.icon}>
            <FontAwesomeIcon icon={solid('clock')} />
          </div>
          <div className={styles.title}>שירות ותמיכה מסביב לשעון</div>
        </Col>
      </Row>
      <MotionRow animate={controls} className={styles.empty}>
        <Col>{message}</Col>
      </MotionRow>
    </Container>
  );
};

export default AdvantagePoints;
