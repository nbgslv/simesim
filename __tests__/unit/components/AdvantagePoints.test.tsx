import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import AdvantagePoints from '../../../components/AdvantagePoints/AdvantagePoints';

describe('AdvantagePoints', () => {
  it('matches snapshot', () => {
    const { container } = render(<AdvantagePoints />);
    expect(container).toMatchSnapshot();
  });

  it('displays message when mouse is over first column', () => {
    const { getByText, getByTestId } = render(<AdvantagePoints />);
    const firstColumn = getByTestId('first-column');
    fireEvent.mouseEnter(firstColumn);
    expect(
      getByText('רכשתם חבילה והטלפון לא תומך? קבלו החזר כספי בלי שאלות')
    ).toBeInTheDocument();
  });

  it('hides message when mouse leaves first column', () => {
    const { getByTestId, queryByText } = render(<AdvantagePoints />);
    const secondColumn = getByTestId('first-column');
    fireEvent.mouseEnter(secondColumn);
    fireEvent.mouseLeave(secondColumn);
    expect(
      queryByText('רכשתם חבילה והטלפון לא תומך? קבלו החזר כספי בלי שאלות')
    ).not.toBeInTheDocument();
  });

  it('displays message when mouse is over second column', () => {
    const { getByText, getByTestId } = render(<AdvantagePoints />);
    const firstColumn = getByTestId('second-column');
    fireEvent.mouseEnter(firstColumn);
    expect(
      getByText(
        'כל הפרטים באתר מוצפנים ומועברים בצורה מאובטחת בטכנולוגיית SSL. כל עסקאות האשראי מאובטחות בתקן PCI המתקדם'
      )
    ).toBeInTheDocument();
  });

  it('hides message when mouse leaves second column', () => {
    const { getByTestId, queryByText } = render(<AdvantagePoints />);
    const secondColumn = getByTestId('second-column');
    fireEvent.mouseEnter(secondColumn);
    fireEvent.mouseLeave(secondColumn);
    expect(
      queryByText(
        'כל הפרטים באתר מוצפנים ומועברים בצורה מאובטחת בטכנולוגיית SSL. כל עסקאות האשראי מאובטחות בתקן PCI המתקדם'
      )
    ).not.toBeInTheDocument();
  });

  it('displays message when mouse is over third column', () => {
    const { getByText, getByTestId } = render(<AdvantagePoints />);
    const firstColumn = getByTestId('third-column');
    fireEvent.mouseEnter(firstColumn);
    expect(
      getByText(
        'מגוון חבילות חו"ל רחב במחירים הנמוכים ביותר; יותר מ-100 מדינות נתמכות; מעבר בין מדינות בלי להחליף eSim!'
      )
    ).toBeInTheDocument();
  });

  it('hides message when mouse leaves third column', () => {
    const { getByTestId, queryByText } = render(<AdvantagePoints />);
    const secondColumn = getByTestId('third-column');
    fireEvent.mouseEnter(secondColumn);
    fireEvent.mouseLeave(secondColumn);
    expect(
      queryByText(
        'מגוון חבילות חו"ל רחב במחירים הנמוכים ביותר; יותר מ-100 מדינות נתמכות; מעבר בין מדינות בלי להחליף eSim!'
      )
    ).not.toBeInTheDocument();
  });

  it('displays message when mouse is over fourth column', () => {
    const { getByText, getByTestId } = render(<AdvantagePoints />);
    const firstColumn = getByTestId('fourth-column');
    fireEvent.mouseEnter(firstColumn);
    expect(
      getByText("גישה לחבילות ותמיכה בוואצאפ, בצ'אט ובטלפון מסביב לשעון")
    ).toBeInTheDocument();
  });

  it('hides message when mouse leaves fourth column', () => {
    const { getByTestId, queryByText } = render(<AdvantagePoints />);
    const secondColumn = getByTestId('fourth-column');
    fireEvent.mouseEnter(secondColumn);
    fireEvent.mouseLeave(secondColumn);
    expect(
      queryByText("גישה לחבילות ותמיכה בוואצאפ, בצ'אט ובטלפון מסביב לשעון")
    ).not.toBeInTheDocument();
  });
});
