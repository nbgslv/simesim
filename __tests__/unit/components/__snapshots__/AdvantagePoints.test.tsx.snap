// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvantagePoints matches snapshot 1`] = `
<div>
  <div
    class="main container"
  >
    <div
      class="mb-2 row"
      layout="position"
    >
      <div
        class="column col"
        data-testid="first-column"
      >
        <div
          class="icon"
        />
        <div
          class="title"
        >
          החזר כספי מובטח
        </div>
      </div>
      <div
        class="column col"
        data-testid="second-column"
      >
        <div
          class="icon"
        />
        <div
          class="title"
        >
          אבטחת המידע שלכם
        </div>
      </div>
      <div
        class="column col"
        data-testid="third-column"
      >
        <div
          class="icon"
        />
        <div
          class="title"
        >
          מגוון חבילות חו"ל רחב
        </div>
      </div>
      <div
        class="column col"
        data-testid="fourth-column"
      >
        <div
          class="icon"
        />
        <div
          class="title"
        >
          שירות ותמיכה מסביב לשעון
        </div>
      </div>
    </div>
    <div
      class="empty row"
    >
      <div
        class="col"
      />
    </div>
  </div>
</div>
`;
